akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ClusterSharding.scala

```diff
@@ -459,7 +459,8 @@ private[akka] class ClusterShardingGuardian extends Actor {
         val encName = URLEncoder.encode(typeName, ByteString.UTF_8)
         val cName = coordinatorSingletonManagerName(encName)
         val cPath = coordinatorPath(encName)
-        val shardRegion = context.child(encName).getOrElse {
+        val shardRegion = context.child(cName).flatMap(_ ⇒ context.child(encName)).getOrElse {
```

Comment:
We must do something else. The problem is that regions and proxies are using the exact same name so it's currently not possible to start both. At least it will not work as intended if proxy is started first and the region afterwards. Then the below `context.actorOf` with `encName` will fail because the proxy with that name is already started and then the region will not be started at all.

I think we have to use a different name for the proxies, such as appending `"Proxy"` to the name of the proxies.

For background context; it was never intended that both proxy and region would be started on same node.